### ISSUE TITLE
Generate documentation from doc comments

### ISSUE THREAD

**Issue by bend-n** [godot-rust/gdext#178]
[gdscript now has doccomments](https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_documentation_comments.html), so it would be neat, if, say
```rs
/// oh my god this is a [b]awesome[/b] function!
/// see also [member cool] and [constant E]
fn on_selected(index: i32) { }
```
would then generate documentation that you can view in the editor:

![image](https://user-images.githubusercontent.com/70787919/224658705-8144734d-22e3-45d4-a58f-b6c80cc5b1e9.png)

or you could even take it a step further, and convert markdown into bbcode so that it meshes better with the rust system.

**Comment by Bromeon** [comment]
> or you could even take it a step further, and convert markdown into bbcode so that it meshes better with the rust system.

This would be the way to go. We could start simple, even supporting formats like `code`, **bold** or _italics_ would already help a lot.

If I remember correctly, `/// text` doc-comments should be visible as `#[doc="text"]` attributes, so they could be picked up by `#[godot_api]`.

**Comment by bend-n** [comment]
Yeah converting basic markdown isnt that hard, it just takes [a couple regexes](https://github.com/bend-n/room2/blob/602ffa0287b2b01d404a618d582203ca105dbaa2/Chat.gd#L32-L41).

**Comment by Bromeon** [comment]
Oh nice!

One thing to keep in mind is that proc-macros are a heavy burden on compile time, and regex replacement can be relatively expensive, especially if it's done "poorly" (i.e. each regex replaces the entire doc-string, causing new allocations and copying). Likely not an issue for small games, this could affect someone with a larger number of documented classes/methods/properties. Would need to be measured though.

**Comment by Dheatly23** [comment]
Just in case if things slow down, at least it should be (default) disabled in debug build. My reasoning is that docs isn't very useful in prototyping stage, and if one wants to release their addons/games it's nice to have docs for end user.

**Comment by Bromeon** [comment]
We do that in gdnative for `async` and `serde`, but I'm not sure if doc-generation is "advanced" or "expensive" enough to warrant extra opt-in from the user. Ideally we get a performant implementation, so this isn't an issue in the first place.

**Comment by sayaks** [comment]
we could maybe use https://crates.io/crates/rustdoc-json for help with parsing the docs similar to rustdoc, however it does rely on the currently unstable [rustdoc JSON backend](https://github.com/rust-lang/rust/issues/76578).

**Comment by lilizoey** [comment]
there isn't a way to do this yet in gdextension, there is a draft PR for it though https://github.com/godotengine/godot/pull/75415

**Comment by StatisMike** [comment]
Passing new PR in this topi, as it seems to be accepted and basically stopped only from 4.2 only because it was in feature-freeze. Most probably it will come in 4.3 release, so we could begin to prepare for it.

https://github.com/godotengine/godot/pull/83747

**Comment by lokimckay** [comment]
The PR StatisMike mentioned is merged, so this PR should be unblocked now
